src/concurrency/genmc/helper.rs

```diff
@@ -101,11 +109,39 @@ pub fn scalar_to_genmc_scalar<'tcx>(
 /// Convert a `GenmcScalar` back into a Miri `Scalar`.
 /// For pointers, attempt to convert the stored base address of their allocation back into an `AllocId`.
 pub fn genmc_scalar_to_scalar<'tcx>(
-    _ecx: &MiriInterpCx<'tcx>,
+    ecx: &MiriInterpCx<'tcx>,
     scalar: GenmcScalar,
     size: Size,
 ) -> InterpResult<'tcx, Scalar> {
-    // FIXME(genmc): Add GenmcScalar to Miri Pointer conversion.
+    if scalar.extra != 0 {
+        // We have a pointer!
```

Comment:
This code is unnecessarily bloated, it can be half as long if you don't let-bind every single thing.